plugin/src/main/java/org/opensearch/ml/profile/MLModelProfile.java

```diff
@@ -79,6 +82,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (memSizeEstimationGPU != null) {
             builder.field("memory_size_estimation_gpu", memSizeEstimationGPU);
         }
+        if (isHidden != null) {
```

Comment:
why isHidden would be null?

Reply:
This is how we build the builders. If the field is present then we put that in the builder. Check the previous code too. 

In this particular scenario isHidden could be null. When the model profiling is being initialized or being used that time we don't put any information about isHidden. We only set when we call the profile api.

Reply:
for boolean, should we give a default when it's read? so when registering the model, we don't know if a model isHidden or not.

Reply:
This is not model registration object, this is model profile object. We set the isHidden value during model registration.